README.md
# Chunk Aligned

This mod make it easier to align stuffs to the chunk grid by tweaking values of power poles and
roboports.

## Power poles & Substations

The following values are modified for power poles, it's generally an upgrade across the board.

|    Mod    |      Type      | Wire Distance  | Coverage |
| --------- | -------------- | -------------- | -------- |
| Vanilla   | Big            | 30 -> 32.25    |          |
| Vanilla   | Substation     | 18 -> 32.25    | 9 -> 16  |
| Extended+ | Big MK2        | 45 -> 48.25    |          |
| Extended+ | Big MK3        | = 64           |          |
| Extended+ | Substation MK2 | 28 -> 48.25    | 14 -> 24 |
| Extended+ | Substation MK3 | 42 -> 64       | 21 -> 32 |
| Krastorio | Substation MK2 | 24.25 -> 48.25 | 12 -> 24 |

The mediums power poles are only modified if the "Align medium power poles" option is checked.

|    Mod    |    Type    | Wire Distance |  Coverage  |
| --------- | ---------- | ------------- | ---------- |
| Vanilla   | Medium     | 9 -> 11.75    | 3.5 -> 4.5 |
| Extended+ | Medium MK2 | 12 -> 16.25   | 4.5 -> 5.5 |
| Extended+ | Medium MK3 | 15 -> 16.25   | 5.5 -> 7.5 |

NOTES:

- Substations are now in the same fast-replace group has big power poles as they have the same wire
  distances.
- Vanilla medium power poles can't be cleanly chunk aligned aligned without being OP, but I tried
  to find a balance.
- Decimals on the wire distances are inspired from Krastorio 2, it allow for a small variations at
  max range.

## Roboports

|    Mod    |             Type             | Logistic Radius | Construction Radius |
| --------- | ---------------------------- | --------------- | ------------------- |
| Vanilla   | Roboport                     | 25 -> 32        | 55 -> 64            |
| Extended+ | Roboport MK2                 | 37.5 -> 48      | 75 -> 96            |
| Extended+ | Roboport MK3                 | 50 -> 64        | 100 -> 128          |
| Krastorio | Small Roboport               | 18 -> 16        | 34 -> 32            |
| Krastorio | Small Roboport (Logis. mode) | 22.95 -> 24     |                     |
| Krastorio | Small Roboport (Const. mode) |                 | 31.875 -> 48        |
| Krastorio | Roboport (Logis. mode)       | 31.875 -> 48    |                     |
| Krastorio | Roboport (Const. mode)       |                 | 68.75 -> 96         |
| Krastorio | Large Roboport               | 100 -> 96       | 200 -> 192          |
| Krastorio | Large Roboport (Logis. mode) | 127.5 -> 128    |                     |
| Krastorio | Large Roboport (Const. mode) |                 | 250 -> 256          |

## License

[Licensed under MIT license](LICENSE.md).


changelog.txt
---------------------------------------------------------------------------------------------------
Version: 1.1.69
Date: 2022-10-14
  Features:
    - Added support for Krastorio2
    - Now with documentated changes 💫💖
  Changes:
    - Removed legacy setting (either don't update or migrate to "Bigger Mediums" setting)
    - Changed setting for "Bigger Mediums" to be a checkbox
    - Tweakeds values (only in positive ways) with inspirations from Krastorio
  Bugfixes:
    - Big power poles are now properly fast replaceable by Substations (was only working
      with FactorioExtended+)
---------------------------------------------------------------------------------------------------
Version: 1.1.0
Date: 2020-11-23
  Changes:
    - Updated to Factorio 1.1.0
---------------------------------------------------------------------------------------------------
Version: 1.0.0
Date: 2020-10-02
  Features:
    - New setting for how medium power poles behave
  Changes:
    - Default settings do no change medium poles anymore
    - There is a deprecated setting for previously used values
---------------------------------------------------------------------------------------------------
Version: 0.2.0
Date: 2020-03-16
  Changes:
    - Add support for LightedPolesPlus power poles
---------------------------------------------------------------------------------------------------
Version: 0.1.1
Date: 2020-03-10
  Changes:
    - Added thumbnail
    - Updated infos
---------------------------------------------------------------------------------------------------
Version: 0.1.0
Date: 2020-03-10
  Features:
    - Power-poles distances now align with chunk grid
    - Roboports ranges now align with chunk grid

updates/vanilla.lua
-- Vanilla entities
local updates = {
    {
        type = "electric-pole",
        name = "big-electric-pole",
        prototypes = {
            maximum_wire_distance = 32.25,
            fast_replaceable_group = "electric-pole-2x2"
        }
    }, {
        type = "electric-pole",
        name = "substation",
        prototypes = {
            maximum_wire_distance = 32.25,
            supply_area_distance = 16,
            fast_replaceable_group = "electric-pole-2x2"
        }
    }, {
        type = "roboport",
        name = "roboport",
        prototypes = { logistics_radius = 32, construction_radius = 64 }
    }
}

-- Try to align medium power poles
if settings.startup['chunk-aligned-align-medium-poles'].value == true then
    table.insert_each(updates, {
        {
            type = "electric-pole",
            name = "medium-electric-pole",
            prototypes = {
                maximum_wire_distance = 11.75,
                supply_area_distance = 4.5
            }
        }
    })
end

return updates


data-final-fixes.lua
-- Update one prototype only if available
local function update_data(type, name, prototypes)
    if (data.raw[type] == nil) then return false end
    if (data.raw[type][name] == nil) then return false end

    log(string.format("[Chunk-Aligned] Updating %s.%s", type, name))
    for k, v in pairs(prototypes) do
        log(string.format("[Chunk-Aligned]     - '%s':  %s -> %s", k, data.raw[type][name][k], v))
        data.raw[type][name][k] = v
    end

    return true
end

-- Update multiple prototypes
local function process_updates(updates)
    for _, update in ipairs(updates) do
        update_data(update.type, update.name, update.prototypes)
    end
end

-- Define the insert_each function in table
function table.insert_each(self, iterable)
    for _, v in ipairs(iterable) do table.insert(self, v) end
end

local vanilla_updates = require("updates.vanilla")
process_updates(vanilla_updates)

local extented_updates = require("updates.factorio_extended")
process_updates(extented_updates)

local krastorio_updates = require("updates.krastorio2")
process_updates(krastorio_updates)

-- Update lighted power poles
require("updates.lighted_poles_plus")


updates/krastorio2.lua
-- Krastorio2 entities
local updates = {
    {
        type = "electric-pole",
        name = "kr-substation-mk2",
        prototypes = {
            maximum_wire_distance = 48.25,
            supply_area_distance = 24,
            fast_replaceable_group = "electric-pole-2x2"
        },
    },
    {
        type = "roboport",
        name = "kr-small-roboport",
        prototypes = { logistics_radius = 16, construction_radius = 32 }
    },
    {
        type = "roboport",
        name = "kr-small-roboport-logistic-mode",
        prototypes = { logistics_radius = 24, construction_radius = 0 }
    },
    {
        type = "roboport",
        name = "kr-small-roboport-construction-mode",
        prototypes = { logistics_radius = 0, construction_radius = 48 }
    },
    {
        type = "roboport",
        name = "roboport-logistic-mode",
        prototypes = { logistics_radius = 48, construction_radius = 0 }
    },
    {
        type = "roboport",
        name = "roboport-construction-mode",
        prototypes = { logistics_radius = 0, construction_radius = 96 }
    },
    {
        type = "roboport",
        name = "kr-large-roboport",
        prototypes = { logistics_radius = 96, construction_radius = 192 }
    },
    {
        type = "roboport",
        name = "kr-large-roboport-logistic-mode",
        prototypes = { logistics_radius = 128, construction_radius = 0 }
    },
    {
        type = "roboport",
        name = "kr-large-roboport-construction-mode",
        prototypes = { logistics_radius = 0, construction_radius = 256 }
    },
}

return updates


updates/factorio_extended.lua
-- Factorio-Extented-Plus entities
local updates = {
    {
        type = "electric-pole",
        name = "big-electric-pole-mk2",
        prototypes = {
            maximum_wire_distance = 48.25,
            fast_replaceable_group = "electric-pole-2x2"
        }
    }, {
        type = "electric-pole",
        name = "big-electric-pole-mk3",
        prototypes = {
            maximum_wire_distance = 64,
            fast_replaceable_group = "electric-pole-2x2"
        }
    }, {
        type = "electric-pole",
        name = "substation-mk2",
        prototypes = {
            maximum_wire_distance = 48.25,
            supply_area_distance = 24,
            fast_replaceable_group = "electric-pole-2x2"
        }
    }, {
        type = "electric-pole",
        name = "substation-mk3",
        prototypes = {
            maximum_wire_distance = 64,
            supply_area_distance = 32,
            fast_replaceable_group = "electric-pole-2x2"
        }
    }, {
        type = "roboport",
        name = "roboport-mk2",
        prototypes = { logistics_radius = 48, construction_radius = 96 }
    }, {
        type = "roboport",
        name = "roboport-mk3",
        prototypes = { logistics_radius = 64, construction_radius = 128 }
    }
}

-- Try to align medium power poles
if settings.startup['chunk-aligned-align-medium-poles'].value == true then
    table.insert_each(updates, {
        {
            type = "electric-pole",
            name = "medium-electric-pole-mk2",
            prototypes = {
                maximum_wire_distance = 16.25,
                supply_area_distance = 5.5
            }
        }, {
            type = "electric-pole",
            name = "medium-electric-pole-mk3",
            prototypes = {
                maximum_wire_distance = 16.25,
                supply_area_distance = 7.5
            }
        }
    })
end

return updates
